src/test/java/teammates/test/cases/browsertests/InstructorEditStudentFeedbackPageUiTest.java

```diff
@@ -82,21 +85,77 @@ private void testAddResponse() throws Exception {
+        FeedbackResponseAttributes frMcq = BackDoor.getFeedbackResponse(
+                fqMcq.getId(), "student1InIESFPTCourse@gmail.tmt", "student1InIESFPTCourse@gmail.tmt");
 
         assertEquals("4", fr.getResponseDetails().getAnswerString());
+        assertNotNull(frMcq);
 
         // Full HTML verification already done in InstructorFeedbackSubmitPageUiTest
         submitPage.verifyHtmlMainContent("/instructorEditStudentFeedbackPageModified.html");
         submitPage.verifyAndCloseSuccessfulSubmissionModal();
     }
 
+    private void testCommentsAction() throws IOException {
+        testCommentsOnMcqQuestion();
+    }
+
+    private void testCommentsOnMcqQuestion() throws IOException {
+        ______TS("add comment on question without response: no effect");
+
+        logout();
+        submitPage = loginToInstructorEditStudentFeedbackPage(
+                "IESFPTCourseinstr", "student1InIESFPTCourse@gmail.tmt", "session1InIESFPTCourse");
+        submitPage.waitForPageToLoad();
+        submitPage.chooseMcqOption(3, 0, "It's good");
+        submitPage.addFeedbackResponseComment("-0-1-3", "Comment without response");
+
+        ______TS("add new comments on question with responses");
+
+        submitPage = loginToInstructorEditStudentFeedbackPage(
+                "IESFPTCourseinstr", "student1InIESFPTCourse@gmail.tmt", "session1InIESFPTCourse");
+        submitPage.waitForPageToLoad();
+        submitPage.verifyHtmlMainContent("/instructorEditStudentFeedbackPageModifiedNoComments.html");
+
+        submitPage.chooseMcqOption(3, 0, "It's perfect"); // select option
+        submitPage.addFeedbackResponseComment("-0-1-3", "New MCQ Comment 1");
+
+        submitPage.submitWithoutConfirmationEmail();
+        submitPage.verifyStatus(Const.StatusMessages.FEEDBACK_RESPONSES_SAVED);
+
+        ______TS("edit comment on question with responses");
+
+        submitPage = loginToInstructorEditStudentFeedbackPage(
+                "IESFPTCourseinstr", "student1InIESFPTCourse@gmail.tmt", "session1InIESFPTCourse");
+        submitPage.waitForPageToLoad();
+        submitPage.verifyHtmlMainContent("/instructorEditStudentFeedbackPageModifiedAddComment.html");
+
+        submitPage.editFeedbackResponseComment("-0-1-3-1", "Edited MCQ Comment 1");
+        submitPage.submitWithoutConfirmationEmail();
+        submitPage.verifyStatus(Const.StatusMessages.FEEDBACK_RESPONSES_SAVED);
+
+        ______TS("delete comment on question with responses");
+
+        submitPage = loginToInstructorEditStudentFeedbackPage(
+                "IESFPTCourseinstr", "student1InIESFPTCourse@gmail.tmt", "session1InIESFPTCourse");
+        submitPage.waitForPageToLoad();
+        submitPage.verifyHtmlMainContent("/instructorEditStudentFeedbackPageModifiedEditComment.html");
+
+        submitPage.deleteFeedbackResponseComment("-0-1-3-1");
+        submitPage.verifyRowMissing("-0-1-6-1");
```

Comment:
Just wondering, why is "-0-1-6-1" used here when the previous line uses "-0-1-3-1"?